src/components/connect.js

```diff
@@ -192,7 +196,14 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
         this._unmounted = true
       }
 
+      shouldComponentUpdate(nextProps, nextState) {
+        return !options.pure ||
+          this.state.data != nextState.data ||
+          (dependsOnProps ? (this._propsDidChange || this.props.children != nextProps.children) : !shallowEqual(this.props, nextProps))
```

Comment:
When I was experimenting before with `children`, I think that `this.props.children != nextProps.children` might always be `true`. I probably don't understand the intricacies of when this it changes well enough, but I remember it not being equal when nothing was changing. Also, just wondering about the intention here. It is correct to re-render a parent when the children change?